spec/method_spec.rb

```diff
@@ -208,6 +208,49 @@ def borscht() paella end
       m = Pry::Method.from_binding(o.borscht)
       expect(m.source).to eq Pry::Method(o.method(:paella)).source
     end
+
+    it 'should not find a wrong method by matching on nil source location' do
+      included_module = Module.new do
+        def self.included(base)
+          base.send :alias_method, "respond_to_without_variables?", "respond_to?"
+          base.send :alias_method, "respond_to?", "respond_to_with_variables?"
+        end
+
+        def respond_to_with_variables?(sym, include_priv=false)
+          respond_to_without_variables?(sym, include_priv)
+        end
+      end
+
+      o = Object.new
+      class << o
+        attr_reader :tasks
+        def task(name, &block)
+          @tasks ||= {}
+          @tasks[name] = block
+        end
+
+        def load
+          path = File.expand_path("../test_code.rb", __FILE__)
```

Comment:
I ran this spec on master and it's not failing. 

So, I checked and I think this is because there is an early return on #find_renamed_method that checks if the file exists.  https://github.com/pry/pry/blob/master/lib/pry/method/weird_method_locator.rb#L157 
 
We could create a file under spec/fixtures for this particular example. What do you think?

Reply:
Oh, that makes sense, I originally had a separate file and removed it while trying to trim down the test example (and must have not red-greened it after that change). I'll move it back into a file.